web/src/components/tasks/anytime-action/tax-clearance-certificate/AnytimeActionTaxClearanceCertificateElement.test.tsx

```diff
@@ -668,6 +674,34 @@ describe("<AnyTimeActionTaxClearanceCertificateReviewElement />", () => {
     describe("renders data when input is provided", () => {});
   });
 
+  it("makes the api post request", async () => {
+    mockApi.postTaxClearanceCertificate.mockResolvedValue({
+      certificatePdfArray: [],
+    });
+    const business = generateBusiness({ id: "Faraz" });
+    const userData = generateUserDataForBusiness(business);
+    renderComponent({ userData });
+    fireEvent.click(screen.getByTestId("stepper-2"));
+    expect(screen.getByTestId("review-tab")).toBeInTheDocument();
+    fireEvent.click(screen.getByTestId("next-button"));
+    await waitFor(() => {
+      expect(mockApi.postTaxClearanceCertificate).toHaveBeenCalledWith(userData);
+    });
+  });
+
+  it("renders the download page when the api post request is successful", async () => {
+    mockApi.postTaxClearanceCertificate.mockResolvedValue({
+      certificatePdfArray: [],
+    });
+    renderComponent({});
+    fireEvent.click(screen.getByTestId("stepper-2"));
+    fireEvent.click(screen.getByTestId("next-button"));
```

Comment:
If we can avoid using `getByTestId` we should. Something we had [discussed briefly in Slack a while back](https://njcio.slack.com/archives/C04J8MTJNS2/p1726592403454379).

Reply:
This work will be picked up immediately after merging this ticket.